src/github.com/matrix-org/dendrite/clientapi/auth/storage/accounts/storage.go

```diff
@@ -93,6 +104,56 @@ func (d *Database) CreateAccount(localpart, plaintextPassword string) (*authtype
 	return d.accounts.insertAccount(localpart, hash)
 }
 
+// PartitionOffsets implements common.PartitionStorer
+func (d *Database) PartitionOffsets(topic string) ([]common.PartitionOffset, error) {
+	return d.partitions.SelectPartitionOffsets(topic)
+}
+
+// SetPartitionOffset implements common.PartitionStorer
+func (d *Database) SetPartitionOffset(topic string, partition int32, offset int64) error {
+	return d.partitions.UpsertPartitionOffset(topic, partition, offset)
+}
+
+// SaveMembership saves the user matching a given localpart as a member of a given
+// room. It also stores the ID of the `join` membership event.
+// If a membership already exists between the user and the room, or of the
+// insert fails, returns the SQL error
+func (d *Database) SaveMembership(localpart string, roomID string, eventID string, txn *sql.Tx) error {
+	return d.memberships.insertMembership(localpart, roomID, eventID, txn)
+}
+
+// RemoveMembershipsByEventIDs removes the memberships of which the `join` membership
+// event ID is included in a given array of events IDs
+// If the removal fails, or if there is no membership to remove, returns an error
+func (d *Database) RemoveMembershipsByEventIDs(eventIDs []string, txn *sql.Tx) error {
+	return d.memberships.deleteMembershipsByEventIDs(eventIDs, txn)
+}
+
+// StartTransaction begins a new SQL transaction and returns it
+// If there was an error during the transaction initialisation, returns it
+func (d *Database) StartTransaction() (*sql.Tx, error) {
+	return d.db.Begin()
+}
+
+// EndTransation is called at the end of a transaction started with StartTransaction
+// If called with an error, the transaction will rollback, if called with nil the
+// transaction will commit
+// If there was an error during either rollback or commit, returns it
+func (d *Database) EndTransation(txn *sql.Tx, err error) error {
+	if err != nil {
+		return txn.Rollback()
+	}
+	return txn.Commit()
+}
+
+// GetMembershipByEventID returns the membership (as a user localpart and a room ID)
+// for which the `join` membership event ID matches a given event ID
+// If no membership match this event ID, the localpart and room ID will be empty strings
+// If an error happens during the retrieval, returns the SQL error
+func (d *Database) GetMembershipByEventID(eventID string) (string, string, error) {
+	return d.memberships.selectMembershipByEventID(eventID)
+}
```

Comment:
I don't think GetMembershipByEventID is used anywhere.  So we can probably get rid of it now.